apps/qubit/modules/settings/actions/diacriticsAction.class.php

```diff
@@ -0,0 +1,104 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+class SettingsDiacriticsAction extends SettingsEditAction
+{
+    // Arrays not allowed in class constants
+    public static $NAMES = [
+        'diacritics',
+    ];
+
+    public function earlyExecute()
+    {
+        parent::earlyExecute();
+
+        $this->updateMessage = $this->i18n->__('Diacritics settings saved.');
+
+        $this->settingDefaults = [
+            'diacritics' => QubitSetting::getByName('diacritics')->getValue(['sourceCulture' => true]),
+        ];
+    }
+
+    public function execute($request)
+    {
+        DefaultEditAction::execute($request);
+
+        if ($request->isMethod('post')) {
+            $this->form->bind($request->getPostParameters(), $request->getFiles());
+
+            if ($this->form->isValid()) {
+                // Check that a diacritics mapping yaml file is uploaded when enabling
+                if (0 === intval($request->getPostParameters()['diacritics']) || (1 === intval($request->getPostParameters()['diacritics']) && 0 !== $request->getFiles()['mappings']['size'])) {
+                    $this->processForm($request);
+                    $this->uploadDiacritics($request);
+
+                    $this->getUser()->setFlash('notice', $this->context->i18n->__('Diacritics setting updated. Rebuild the search index for the changes to be applied.'));
+                    $this->redirect(['module' => 'settings', 'action' => 'diacritics']);
+                } else {
+                    $this->getUser()->setFlash('error', $this->context->i18n->__('To enable diacritics setting, please upload a char_filter mapping YAML.'));
+                    $this->redirect(['module' => 'settings', 'action' => 'diacritics']);
+                }
+            }
+        }
+    }
+
+    public function processForm($request)
+    {
+        sfConfig::set('app_diacritics', $request->getPostParameters()['diacritics']);
+        QubitSetting::findAndSave('diacritics', $request->getPostParameters()['diacritics'], ['sourceCulture' => true]);
+    }
+
+    public function uploadDiacritics($request)
+    {
+        $file = $request->getFiles('mappings');
+        $diacriticsMappingPath = sfConfig::get('sf_plugins_dir').DIRECTORY_SEPARATOR.'arElasticSearchPlugin'.DIRECTORY_SEPARATOR.'config'.DIRECTORY_SEPARATOR.'diacritics_mapping.yml';
+
+        if (1 === (int) $request->getPostParameters()['diacritics']) {
+            if ('application/x-yaml' === $request->getFiles()['mappings']['type']) {
+                // Move uploaded file to replace diacritics_mapping.yml.
+                try {
+                    $file = Qubit::moveUploadFile($file);
+                } catch (sfException $e) {
+                    $this->getUser()->setFlash('error', $e->getMessage());
```

Comment:
It might be better, for security reasons, to provide a generic error like "Error: unable to upload" or some such thing. Otherwise the error message might divulge system information.

Reply:
It should also return after setting the flash, right?